src/solvers/base_solver.cpp

```diff
@@ -152,20 +142,18 @@ double BaseSolver::GetTime() const { return m_time; }
 
 int BaseSolver::GetCycle() const { return m_cycle; }
 
-void BaseSolver::InitializeOutput(const OutputType output_type, std::string root_name, std::vector<std::string> names)
+void BaseSolver::InitializeOutput(const OutputType output_type, std::string root_name)
 {
-  MFEM_ASSERT(names.size() == m_state.size(), "State vector and name arrays are not the same size.");
-
   m_root_name = root_name;
 
   m_output_type = output_type;
 
   switch (m_output_type) {
     case OutputType::VisIt: {
-      m_visit_dc = new mfem::VisItDataCollection(m_root_name, m_state.begin()->mesh);
+      m_visit_dc = std::make_unique<mfem::VisItDataCollection>(m_root_name, m_state.begin()->mesh.get());
 
-      for (unsigned int i = 0; i < names.size(); i++) {
-        m_visit_dc->RegisterField(names[i], m_state[i].gf.get());
+      for (unsigned int i = 0; i < m_state.size(); i++) {
```

Comment:
You could be even fancier:

for(const auto& state: m_state) {
  m_visit_dc->RegisterField(state.name, state.gf.get());
}

Should work with the caveat I didn't actually try compiling it.

Also "gf" will always make me think of girlfriend and "state.gf.get()" makes me chuckle.